uaclient/api/u/pro/services/list/v1.py

```diff
@@ -0,0 +1,206 @@
+    fields = [
+        Field("services", data_list(ServiceInfo), doc="List of services"),
+    ]
+
+    def __init__(self, *, services: List[ServiceInfo]):
+        self.services = services
+
+
+def get_service_list() -> ServiceListResult:
+    return _get_service_list(UAConfig())
+
+
+def _get_service_list(cfg: UAConfig) -> ServiceListResult:
+    if not _is_attached(cfg).is_attached:
+        return unattached_services(cfg)
+    return attached_services(cfg)
+
+
+def attached_services(cfg: UAConfig) -> ServiceListResult:
+    services_list = []
+    machine_token_file = machine_token.get_machine_token_file(cfg)
+    resources = machine_token_file.machine_token.get("availableResources")
+    if not resources:
+        resources = get_available_resources(cfg)
+    inapplicable_resources = {
+        resource["name"]: resource.get("description")
+        for resource in sorted(resources, key=lambda x: x.get("name", ""))
+        if not resource.get("available")
+    }
+    for resource in resources:
+        try:
+            ent = entitlement_factory(cfg=cfg, name=resource.get("name", ""))
+        except exceptions.EntitlementNotFoundError:
+            LOG.debug(
+                "Ignoring availability of unknown service %s from contract "
+                "server",
+                resource.get("name", "without a 'name' key"),
+            )
+            continue
+
+        services_list.append(
+            _get_service_info(cfg, ent, inapplicable_resources)
+        )
+    services_list.sort(key=lambda x: x.name)
+    return ServiceListResult(services=services_list)
+
+
+def unattached_services(cfg: UAConfig) -> ServiceListResult:
+    services_list = []
+    resources = get_available_resources(cfg)
+    for resource in resources:
+        available = True if resource.get("available") else False
+        try:
+            ent = entitlement_factory(cfg=cfg, name=resource.get("name", ""))
+        except exceptions.EntitlementNotFoundError:
+            LOG.debug(
+                "Ignoring availability of unknown service %s from contract "
+                "server",
+                resource.get("name", "without a 'name' key"),
+            )
+            continue
+        service = ServiceInfo(
+            name=resource.get("presentedAs", resource["name"]),
+            desc=ent.description,
+            available=available,
+            entitled=None,
+        )
+        services_list.append(service)
+    services_list.sort(key=lambda x: x.name)
+    return ServiceListResult(services=services_list)
+
+
+def _get_service_info(
+    cfg: UAConfig, ent: UAEntitlement, inapplicable_resources: Dict
+) -> ServiceInfo:
+    contract_status = ent.contract_status()
+    entitled = None
+    if contract_status == ContractStatus.UNENTITLED:
+        entitled = False
+    elif contract_status == ContractStatus.ENTITLED:
+        entitled = True
+    available = True if ent.name not in inapplicable_resources else False
```

Comment:
I think we should generally be using `ent.applicability_status()` for the value of `available` - right @lucasmoura @renanrodrigo ?

Reply:
I think @dheyay is just replicating the logic we have on status here. However, I think it is worth testing out replacing this logic with `applicability_status`

Reply:
It works with the `applicability_status` logic and does not need the additional inapplicable_resources. Updated the implementation and the endpoint/function names @orndorffgrant